src/NUnitFramework/framework/Constraints/Comparers/IComparer.cs

```diff
@@ -0,0 +1,45 @@
+﻿// ***********************************************************************
+// Copyright (c) 2009 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+
+namespace NUnit.Framework.Constraints.Comparers
+{
+    /// <summary>
+    /// Interface for comparing two <see cref="Object"/>s.
+    /// </summary>
+    internal interface IComparer
+    {
+        /// <summary>
+        /// Method for comparing two objects with a tolerance.
+        /// </summary>
+        /// <param name="x">The first object to compare.</param>
+        /// <param name="y">The second object to compare.</param>
+        /// <param name="tolerance">The tolerance to use when comparing the objects.</param>
+        /// <returns>
+        ///     <c>null</c> if the objects cannot be compared using the method.
+        ///     Otherwise the result of the comparison is returned.
+        /// </returns>
+        bool? Equal(object x, object y, ref Tolerance tolerance);
```

Comment:
Interesting using `bool?` here. When I was thinking of doing this, I was going to add a `CanCompare()` method. This saves the extra cast. 👍